tests/oracle.rs

```diff
@@ -0,0 +1,372 @@
+
+            exec.tx::execute_foreign_procedure
+            # => [STORAGE_VALUE]
+
+            # assert the correctness of the obtained value
+            push.{oracle_data_2} assert_eqw
+            # => []
+
+            ### get oracle data 3 ###
+
+            # pad the stack for the `execute_foreign_procedure`execution
+            padw padw padw push.0.0
+            # => [pad(14)]
+
+            # push the index of desired storage item
+            push.2
+
+            # get the hash of the `get_item` account procedure
+            push.{get_item_foreign_hash}
+
+            # push the foreign account id
+            push.{foreign_account_id}
+            # => [foreign_account_id, FOREIGN_PROC_ROOT, storage_item_index, pad(14)]
+
+            exec.tx::execute_foreign_procedure
+            # => [STORAGE_VALUE]
+
+            # assert the correctness of the obtained value
+            push.{oracle_data_3} assert_eqw
+            # => []
+
+            ### get oracle data 4 ###
+
+            # pad the stack for the `execute_foreign_procedure`execution
+            padw padw padw push.0.0
+            # => [pad(14)]
+
+            # push the index of desired storage item
+            push.3
+
+            # get the hash of the `get_item` account procedure
+            push.{get_item_foreign_hash}
+
+            # push the foreign account id
+            push.{foreign_account_id}
+            # => [foreign_account_id, FOREIGN_PROC_ROOT, storage_item_index, pad(14)]
+
+            exec.tx::execute_foreign_procedure
+            # => [STORAGE_VALUE]
+
+            # assert the correctness of the obtained value
+            push.{oracle_data_4} assert_eqw
+            # => []
+
+            # truncate the stack
+            exec.sys::truncate_stack
+        end
+        ",
+        foreign_account_id = oracle_account.id(),
+        get_item_foreign_hash = oracle_account.code().procedures()[1].mast_root(),
+        oracle_data_1 = &word_to_masm(oracle_data_word_1),
+        oracle_data_2 = &word_to_masm(oracle_data_word_2),
+        oracle_data_3 = &word_to_masm(oracle_data_word_3),
+        oracle_data_4 = &word_to_masm(oracle_data_word_4)
+    );
+
+    let tx_script =
+        TransactionScript::compile(code, vec![], TransactionKernel::testing_assembler()).unwrap();
+
+    let tx_context = mock_chain
+        .build_tx_context(native_account.id(), &[], &[])
+        .advice_inputs(advice_inputs.clone())
+        .tx_script(tx_script)
+        .build();
+
+    let block_ref = tx_context.tx_inputs().block_header().block_num();
+    let note_ids = tx_context
+        .tx_inputs()
+        .input_notes()
+        .iter()
+        .map(|note| note.id())
+        .collect::<Vec<_>>();
+
+    let mut executor: TransactionExecutor =
+        TransactionExecutor::new(Arc::new(tx_context.clone()), None).with_tracing();
+
+    // load the mast forest of the foreign account's code to be able to create an account procedure
```

Comment:
Is it important that we call that mast forest? (It should be MAST forest, right?). Maybe that confuses more than it helps. Can't we simply say 'Load the oracle account code into the transaction"